airflow/kubernetes/pod_launcher.py

```diff
@@ -123,8 +124,13 @@ def run_pod(
     def _monitor_pod(self, pod: V1Pod, get_logs: bool) -> Tuple[State, Optional[str]]:
         if get_logs:
             logs = self.read_pod_logs(pod)
-            for line in logs:
-                self.log.info(line)
+            # Catch exception as for loop on HttpResponse may corrupt.
+            try:
+                for line in logs:
+                    self.log.info(line)
+            except ProtocolError as e:
```

Comment:
I guess this is better than nothing (and better than failing) but would it be possible to provide the `timestamps=True` (which we filter back out again) and use `since_time` to start-off again where we left off.